src/url-params.ts

```diff
@@ -163,10 +172,28 @@ export class AnalyticsUrlParams {
 	/**
 	 * @return full query parameter string that can be appended to URLs
 	 */
-	getQueryString(): string {
-		return [this.getDeviceIdsQueryString(), this.getSessionIdQueryString()]
-			.filter(x => x)
-			.join('&');
+	getQueryString(destinationUrl?: URL, currentUrl?: URL): string {
+		const regex = /([a-zA-Z0-9-]+)(\.[a-zA-Z]{2,3})?(\.[a-zA-Z]+$)/g;
+		const actualDomainMatch = currentUrl
+			? currentUrl.hostname.match(regex)
+			: window.location.hostname.match(regex);
```

Comment:
What happens if the `window` object is not accessible?
Is it going to be `undefined`, throw an exception or do something else? Just trying to cover the logical cases here.

Reply:
@pranasziaukas now I wrote it differently for it to check also if window object exists